### PR TITLE
Add OBB.

### PR DESCRIPTION
**Pull request by feuerste**
We want to extend iterators to take local transformations. In order to implement all of them properly, we need an object oriented bounding box, so we can transform an aabb properly.

### CODE REVIEW

File: src/math.rs [hunk 1]
```diff
@@ -78,7 +201,7 @@ pub struct OrientedBeam {
     translation_inv: Vector3<f64>,
     half_extent: Vector2<f64>,
     corners: [Point3<f64>; 4],
-    separating_axes: [Vector3<f64>; 5],
+    separating_axes: Vec<Vector3<f64>>,
```
Comment (nnmm) [inline src/math.rs]: What's the benefit of a vector?

Comment (nnmm) [reply src/math.rs]: Ah, you're filtering before intersects().

File: src/math.rs [hunk 2]
```diff
@@ -68,6 +71,126 @@ impl Cube {
...
+            unit_x.cross(rot_y).normalize(),
+            unit_x.cross(rot_z).normalize(),
+            unit_y.cross(rot_x).normalize(),
+            unit_y.cross(rot_y).normalize(),
+            unit_y.cross(rot_z).normalize(),
+            unit_z.cross(rot_x).normalize(),
+            unit_z.cross(rot_y).normalize(),
+            unit_z.cross(rot_z).normalize(),
+        ] {
+            let is_finite_and_non_parallel = is_finite(&axis)
```
Comment (nnmm) [inline src/math.rs]: Is the non-parallel condition just an optimization?

File: src/math.rs (hunk 1 above)
Comment (feuerste) [reply src/math.rs]: Yes, so we can avoid the check later.

File: src/math.rs (hunk 2 above)
Comment (feuerste) [reply src/math.rs]: Yes, it's an optimization to get rid of as many axes to check before actually performing the intersection tests.

**Comment by feuerste** [comment]
@wally-the-cartographer merge

**Comment by feuerste** [comment]
Thanks for the review Nikolai!

**Pull request merged**